src/components/App.js

```diff
@@ -3,20 +3,143 @@ import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 import calculate from '../logic/calculate';
 
-function App() {
-  const data = { total: '20', next: '5', operation: '+' };
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      decimal: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-  const buttonname = '+/-';
+  handleClick(event) {
+    const {
+      next, decimal, total, operation,
+    } = this.state;
+    const calc = calculate(this.state, event.target.innerHTML);
 
-  calculate(data, buttonname);
+    if (calc.operation1 == null && calc.next1 != null) {
+      if (next == null) {
+        if (calc.decimal) {
+          this.setState(() => ({
+            next: calc.next1,
+            decimal: true,
+          }
 
-  return (
-    <>
-      <ButtonPanel />
-      <br />
-      <Display result="25" />
-    </>
-  );
-}
+          ));
+        } else {
+          this.setState(() => ({
+            next: calc.next1,
+          }
+
```

Comment:
- [ ] Please move all the logic from the calculation to a different file, to keep your code organized. Again, you may follow the example from [my solution](https://github.com/voscarmv/calculator/blob/7aa17ac397ff1a3bf53f335d21568d0524d288e4/src/logic/calculate.js#L3).